submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/util/AccNoPatternUtil.kt

```diff
@@ -1,53 +1,54 @@
 package ac.uk.ebi.biostd.submission.util
 
-import ac.uk.ebi.biostd.submission.processors.ACC_PATTERN
-import arrow.core.Option
+import ac.uk.ebi.biostd.submission.exceptions.InvalidAccNoPattern
+import ac.uk.ebi.biostd.submission.exceptions.InvalidPatternException
 import arrow.core.getOrElse
 import arrow.core.orElse
+import ebi.ac.uk.model.AccNumber
 import ebi.ac.uk.model.AccPattern
 import ebi.ac.uk.util.regex.firstGroup
+import ebi.ac.uk.util.regex.match
 import ebi.ac.uk.util.regex.secondGroup
 import ebi.ac.uk.util.regex.thirdGroup
 import java.util.regex.Matcher
-import java.util.regex.Pattern
 
-private val onlyPrefix = ACC_PATTERN.format("([A-Z,-]*),").toPattern()
-private val onlyPostfix = ACC_PATTERN.format(",([A-Z,-]*)").toPattern()
-private val prefixPostfix = ACC_PATTERN.format("([A-Z,-]*),([A-Z,-]*)").toPattern()
-private val extractionPattern = "(\\D*)([0-9]+)(\\D*)".toPattern()
+private const val ACC_PATTERN = "\\!\\{%s\\}"
+private const val EXPECTED_PATTERN = "([A-Z,-]*),([A-Z,-]*)"
 
-class PatternProcessor {
+private val ONLY_PREFIX = ACC_PATTERN.format("([A-Z,-]*),").toPattern()
+private val ONLY_POSTFIX = ACC_PATTERN.format(",([A-Z,-]*)").toPattern()
+private val PREFIX_POSTFIX = ACC_PATTERN.format(EXPECTED_PATTERN).toPattern()
+private val EXTRACTION_PATTERN = "(\\D*)([0-9]+)(\\D*)".toPattern()
 
-    fun generateAccNumber(accPattern: String, sequenceFunction: (AccPattern) -> Long) =
-        getPrefixAccPattern(accPattern)
-            .orElse { getPostfixAccPattern(accPattern) }
-            .orElse { getPrefixPostfixPattern(accPattern) }
-            .map { pattern -> AccNumber(pattern, sequenceFunction(pattern)) }
-            .getOrElse { throw IllegalStateException() }
+class AccNoPatternUtil {
 
-    fun extractAccessNumber(accNo: String) = tryToGet(accNo, extractionPattern) {
-        AccNumber(AccPattern(it.firstGroup(), it.thirdGroup()), it.secondGroup().toLong())
-    }.getOrElse { throw IllegalStateException() }
+    fun getPattern(accPattern: String) =
+            getPrefixAccPattern(accPattern)
+                    .orElse { getPostfixAccPattern(accPattern) }
+                    .orElse { getPrefixPostfixPattern(accPattern) }
+                    .getOrElse { throw InvalidPatternException(accPattern, EXPECTED_PATTERN) }
+
+    /**
```

Comment:
I'd suggest: Determinates if the submission accession number is a pattern, which is determined based on whether or not it matches the...